invenio_stats/templates.py

```diff
@@ -21,3 +23,17 @@ def register_templates():
     ]
 
     return event_templates + aggregation_templates
+
+
+def register_templates():
+    """Register search templates for events."""
+    if current_app.config["STATS_REGISTER_INDEX_TEMPLATES"]:
+        return []
+    return _collect_templates()
+
+
+def register_index_templates():
+    """Register search index templates for events."""
+    if not current_app.config["STATS_REGISTER_INDEX_TEMPLATES"]:
+        return []
```

Comment:
these functions look a bit too similar and I am worried they will be easily confusing, but I don't have a good idea on how to avoid this. Maybe a bit more elaborate comments on both to explain the backwards compatibility of `register_templates` vs `register_index_templates`?
also I guess, we cannot raise errors instead of empty list, because we lose the backwards compatibility ?